teuthology/suite/__init__.py

```diff
@@ -185,6 +188,7 @@ def get_rerun_conf(conf):
                   stored_subset=subset)
     if conf.no_nested_subset is True:
         log.info('Nested subsets disabled')
+        no_nested_subset = conf.no_nested_subset
```

Comment:
no_nested_subset should get replaced with a new value when we explicitly pass the no_nested_subset flag.